sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -140,6 +141,7 @@ case class AnalyzeColumnCommand(
     case DoubleType | FloatType => true
     case BooleanType => true
     case _: DatetimeType => true
+    case CharType(_) | VarcharType(_) if !SQLConf.get.preserveCharVarcharTypeInfo => false
```

Comment:
I don't think we are ready to support char/varchar column statistics, the story is not completed yet. E.g. how do we consider padding when calculating min/max? And how do we push down predicates since Spark does read-side padding by default? The predicate is like `rpad(col, 5) > 'a'`.

I'd prefer to keep char/varchar unsupported for now.